README.md

```diff
@@ -43,153 +43,180 @@ installed on your machine (whichever version is required by Vault).
 
-### Get Plugin
+### Build Plugin
 
-Clone this repository:
+1. Clone this repository:
 
-```text
-$ mkdir $GOPATH/src/github.com/hashicorp/vault-plugin-secrets-gcp`
-$ cd $GOPATH/src/github.com/hashicorp/
-$ git clone https://github.com/hashicorp/vault-plugin-secrets-gcp.git
-```
+    ```sh
+    $ mkdir $GOPATH/src/github.com/hashicorp/vault-plugin-secrets-gcp`
+    $ cd $GOPATH/src/github.com/hashicorp/
+    $ git clone https://github.com/hashicorp/vault-plugin-secrets-gcp.git
+    ```
 
-or use `go get github.com/hashicorp/vault-plugin-secrets-gcp`
+    or use `go get github.com/hashicorp/vault-plugin-secrets-gcp`
 
-You can then download any required build tools by bootstrapping your
+1. You can then download any required build tools by bootstrapping your
 environment:
 
-```sh
-$ make bootstrap
-```
+    ```sh
+    $ make bootstrap
+    ```
 
-To compile a development version of this plugin, run `make` or `make dev`.
+1. To compile a development version of this plugin, run `make` or `make dev`.
 This will put the plugin binary in the `bin` and `$GOPATH/bin` folders. `dev`
 mode will only generate the binary for your platform and is faster:
 
-```sh
-$ make
-$ make dev
-```
-
-### Install Plugin in Vault
-
-Put the plugin binary into a location of your choice. This directory
-will be specified as the [`plugin_directory`](https://www.vaultproject.io/docs/configuration/index.html#plugin_directory)
-in the Vault config used to start the server.
-
-```hcl
-plugin_directory = "path/to/plugin/directory"
-```
+    ```sh
+    $ make
+    $ make dev
+    ```
 
-Start a Vault server with this config file:
+### Start Vault
 
-```sh
-$ vault server -config=path/to/config.json #...
-```
+You will need a Vault instance running locally with a development plugin directory set up.
 
-Once the server is started, register the plugin in the Vault server's [plugin catalog](https://www.vaultproject.io/docs/internals/plugins.html#plugin-catalog):
+1. Start a local Vault server:
 
-```sh
-$ vault write sys/plugins/catalog/gcpsecrets \
-    sha_256="$(shasum -a 256 path/to/plugin/directory/vault-plugin-secrets-gcp | cut -d " " -f1)" \
-    command="vault-plugin-secrets-gcp"
-```
+    ```sh
+    $ vault server -dev -dev-root-token-id=root -dev-plugin-dir="$GOPATH/vault-plugins"
+    ```
+   
+### Set Up Plugin in Vault
 
-Any name can be substituted for the plugin name "gcpsecrets". This
-name will be referenced in the next step, where we enable the secrets
-plugin backend using the GCP secrets plugin:
+Next you want Vault to use the newly built binary instead of the version bundled with Vault.
+You can use the scripted or manual methods depending on your preference.
 
-```sh
-$ vault secrets enable --plugin-name='gcpsecrets' --path="gcp" plugin
-```
+Scripted method.
 
-### Tests
+1. A Terraform project sets up the necessary gcp service account & credentials.
+   You can configure the project id via this environment variable: `TF_VAR_GOOGLE_CLOUD_PROJECT_ID` 
```

Comment:
I just tested this out. If I set `TF_VAR_GOOGLE_CLOUD_PROJECT_ID` terraform asks me to set `var.GOOGLE_CLOUD_PROJECT` after I run `make setup-env`. Is that expected?

Reply:
That's weird, [TF should infer variable default values](https://support.hashicorp.com/hc/en-us/articles/4547786359571-Reading-and-using-environment-variables-in-Terraform-runs) from env vars prefixed with TF_VAR_ and not prompt you to declare a value.

I tried on my side:
```
export TF_VAR_GOOGLE_CLOUD_PROJECT_ID=hc-1234
make setup-env
```

And the plan executed without asking me to define `var.GOOGLE_CLOUD_PROJECT`. Were the 2 commands done in the same context?

Reply:
They were done in the same context. I just double checked my history. But I tried it again just now and it works as expected so... 🤷